katz/src/main/kotlin/katz/data/Eval.kt

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (C) 2017 The Katz Authors
```

Comment:
We should modify all of our copyright headers to include that most of the datatypes like this one is a port originating from Scala cats. We don't want people to think we created this from scratch since the exercise is mainly about porting most of the cats one that are useful for what we are trying to achieve. I don't want people in the Scala community upset about not being properly credited in the copyrights so such a fact should be state clearly everywhere in the project including header files, README and LICENSE files. 

I'm not an attorney but I think it's important that those that have come up with the original encoding of these datatypes in both cats and Scalaz get properly credited.

We should follow similar practices as this:

https://github.com/typelevel/cats/blob/master/COPYING
https://github.com/typelevel/cats/blob/master/AUTHORS.md

Also I think we should remove copyright headers from all the files as a lot of the code on them is not really our copyright. Copyright assignment and contribution should be in the License and Copying files only to avoid mistakes like automatically adding a header to a file whose code is entirely translated from cats or scalaz.